zcash_client_backend/src/data_api.rs

```diff
@@ -1583,11 +1583,20 @@ impl AccountBirthday {
 /// - If [`WalletWrite::import_account_hd`] is used to import accounts with non-sequential
 ///   ZIP 32 account indices from the same seed, a call to [`WalletWrite::create_account`]
 ///   will use the ZIP 32 account index just after the highest-numbered existing account.
-/// - If an account is imported via [`WalletWrite::import_account_ufvk`], and then a later
-///   call to either [`WalletWrite::create_account`] or [`WalletWrite::import_account_hd`]
-///   would produce the same UFVK, an error will be returned.
-///   - A future change to this trait might introduce a method to "upgrade" an imported
-///     account with derivation information. See [zcash/librustzcash#1284] for details.
+/// - If an account is added to the wallet, and then a later call to one of the methods
+///   would produce the same UFVK, an error will be returned. While any combination of
+///   method calls can cause this, there are two main ways it can occur that users of the
+///   `WalletWrite` trait should watch out for:
+///   - An account is created via [`WalletWrite::create_account`] with an auto-selected
+///     ZIP 32 account index, and that index is later imported explicitly via either
+///     [`WalletWrite::import_account_hd`] or [`WalletWrite::import_account_ufvk`].
+///   - An account is imported via [`WalletWrite::import_account_ufvk`], and the ZIP 32
+///     account index corresponding to that account's UFVK is later imported either
+///     implicitly via [`WalletWrite::create_account`] or explicitly via
+///     [`WalletWrite::import_account_hd`].
+///
```

Comment:
```suggestion
///
/// Note that accounts having UFVKs that are not identical but have shared
/// components (for example, two accounts having the same Sapling FVK, one
/// of which also has an Orchard FVK while the other does not) are currently
/// allowed. This will not be allowed in future.
```

@str4d wrote (in response to slightly different wording of the last sentence that said "may not" rather than "will not"):
> NACK, I think we should decide how to handle this, and not in this PR.

Absolutely we should, but this documents the current behaviour correctly. Then in the PR where we fix it (which I agree, should be before the release), we can and should change the comment accordingly.

Reply:
These suggestions are included in #1489, which then goes on to fix it and change the comment.